maroon/src/app.rs

```diff
@@ -1,22 +1,55 @@
 use crate::p2p_interface::{Inbox, NodeState, Outbox, P2PChannels};
 use common::{
+    async_interface::{AsyncInterface, ReqResPair},
     gm_request_response::Transaction,
     range_key::{self, KeyOffset, KeyRange, TransactionID},
 };
 use libp2p::PeerId;
-use log::{debug, error, info, warn};
+use log::{error, info};
 use std::{
     collections::{HashMap, HashSet},
     num::NonZeroUsize,
     time::Duration,
 };
 use tokio::sync::oneshot;
 
-pub struct App {
+pub enum Request {
+    GetState,
+}
+#[derive(Debug, PartialEq, Eq)]
+pub enum Response {
+    State(CurrentOffsets),
+}
+
+#[derive(Debug, PartialEq, Eq)]
+pub struct CurrentOffsets {
+    pub self_offsets: HashMap<KeyRange, KeyOffset>,
+    pub consensus_offset: HashMap<KeyRange, KeyOffset>,
+}
+
+#[derive(Clone, Copy, Debug)]
+pub struct Params {
+    /// how often node will send state info to other nodes
+    pub advertise_period: std::time::Duration,
     /// minimum amount of nodes that should have the same transactions(+ current one) in order to confirm them
-    consensus_nodes: NonZeroUsize,
+    pub consensus_nodes: NonZeroUsize,
+}
+
+impl Params {
+    pub fn default() -> Params {
+        Params {
+            advertise_period: Duration::from_secs(5),
+            consensus_nodes: NonZeroUsize::new(2).unwrap(),
```

Comment:
Another nit: We should probably find a way to make it clear on the code level that this is not `etcd`-grade consensus. While I know these are two different things, I did cringe a bit on `2` being an even number here.

Perhaps call it some `min_replication_factor`, or even `min_per_rollup_replication_factor`?